tests/integration/p2p.go

```diff
@@ -307,8 +313,14 @@ func configureReplicator(
 	err := sourceNode.SetReplicator(s.ctx, client.Replicator{
 		Info: targetNode.PeerInfo(),
 	})
-	require.NoError(s.t, err)
-	setupReplicatorWaitSync(s, 0, cfg, sourceNode, targetNode)
+
+	if err == nil {
+		require.Equal(s.t, cfg.ExpectedError, "")
```

Comment:
todo: This is incorrect and does not match the documentation on `cfg.ExpectedError`, or the behaviour of every other `action.ExpectedError` property.  Please redo this function to match the rest of the test suite.

Reply:
Which documentation does it not match? Are you referring to the string can be partial? It will do that check in the following lines?

Reply:
Ah I didn't read it properly and assumed you were checking an error string.  Could you not just remove the require and check this after the `assertExpectedErrorRaised` call?  The behaviour is still different to the other `ExpectedError`s as you have a require instead of an assert, and it is outside of the function we use to do just this (duplication of consistency enforcing code).

```
expectedErrorRaised := AssertError(s.t, s.testCase.Description, err, cfg.ExpectedError)
assertExpectedErrorRaised(s.t, s.testCase.Description, cfg.ExpectedError, expectedErrorRaised)
if err == nil {
  setupReplicatorWaitSync(s, 0, cfg, sourceNode, targetNode)
}
```

Reply:
And if you really want/need a require check there, please use `require.NoError` instead of equal-empty-string, it is much more readable, and will give a nicer failure when running the test

Reply:
I am guessing you mean that I should:

Change this: 
```
	err := sourceNode.SetReplicator(s.ctx, client.Replicator{
		Info: targetNode.PeerInfo(),
	})

	if err == nil {
		require.Equal(s.t, cfg.ExpectedError, "")
		setupReplicatorWaitSync(s, 0, cfg, sourceNode, targetNode)
	}

	expectedErrorRaised := AssertError(s.t, s.testCase.Description, err, cfg.ExpectedError)
	assertExpectedErrorRaised(s.t, s.testCase.Description, cfg.ExpectedError, expectedErrorRaised)
```

To:
```
	err := sourceNode.SetReplicator(s.ctx, client.Replicator{
		Info: targetNode.PeerInfo(),
	})
	expectedErrorRaised := AssertError(s.t, s.testCase.Description, err, cfg.ExpectedError)
	assertExpectedErrorRaised(s.t, s.testCase.Description, cfg.ExpectedError, expectedErrorRaised)
	setupReplicatorWaitSync(s, 0, cfg, sourceNode, targetNode)
```

Reply:
Yes, roughly, you may still need the `if` though - `assertExpectedErrorRaised` asserts, and so may still call `setupReplicatorWaitSync` if you are not careful.  This way around will ensure consistent behaviour RE our other test errors.

Reply:
- [x] Accept todo 

Behavior is still same IMO before this todo, but I did the todo anyways.